cra-server/router/routes.js

```diff
@@ -241,10 +245,9 @@ router.get("/open-metadata/admin-services/*", (req, res) => {
     method: "get",
     url: urlRoot + incomingPath,
     httpsAgent: new https.Agent({
-      // ca: - at some stage add the certificate authority
-      cert,
-      key,
-      rejectUnauthorized: false,
+      ca: truststore,
+      pfx: keystore,
```

Comment:
Can you point me to the right axios doc? I'm not sure how to validate this. pfx is a format, but presumably here the context is it's use as a certiicate to optionally send? But if so how does it know which key to use? Also you have both a truststore and a keystore, each of which could have different passphrases -- and I assume the API would account for this.

Does this code work, or is it your interpretation from the docs? My feeling is it wouldn't work?

Reply:
@planetf1 the code works, meaning it solves the codeql issue, and it works in the UI.

Reply:
Ok - I couldn't find a clear description of the API to verify. But there's something missing. A .pfx can contain multiple certs. If so the code needs to know which to use. We may need to clarify docs/limitations. It may also be it happens to work with the current certs, but when I replace them it may not.

Also how, in this call, is it dealing with the case where the passwords to the two .p12 files differ -- which might well happen as one is more 'secret' than the other. Does it assume no password on the "truststore"?

I'll do the change to replace the certs (locally) soon and will retest

Reply:
I have struggled to find good docs for this . I found https://nodejs.org/api/https.html#https_class_https_agent.

Reply:
Thanks - I found that but it felt incomplete without 'playing' with the API ... I think that's the only route....

Reply:
the options point to https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener for the pfx .

Reply:
In those docs I see options <Object> Accepts options from tls.createServer(), tls.createSecureContext() and see the definition of ca and pfx. ca is being used to override the trusted CA certificates, this is saying what to trust I think,  so there is no associated passphrase. The passphrase and pfx options are a pair.

ca <string> | <string[]> | <Buffer> | <Buffer[]> Optionally override the trusted CA certificates. Default is to trust the well-known CAs curated by Mozilla. Mozilla's CAs are completely replaced when CAs are explicitly specified using this option. The value can be a string or Buffer, or an Array of strings and/or Buffers. Any string or Buffer can contain multiple PEM CAs concatenated together. The peer's certificate must be chainable to a CA trusted by the server for the connection to be authenticated. When using certificates that are not chainable to a well-known CA, the certificate's CA must be explicitly specified as a trusted or the connection will fail to authenticate. If the peer uses a certificate that doesn't match or chain to one of the default CAs, use the ca option to provide a CA certificate that the peer's certificate can match or chain to. For self-signed certificates, the certificate is its own CA, and must be provided. For PEM encoded certificates, supported types are "TRUSTED CERTIFICATE", "X509 CERTIFICATE", and "CERTIFICATE". See also tls.rootCertificates.